### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by JonasKruckenberg**
Just a small bump to dependencies to fix my "duplicate dependency ocd" 😄

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,20 +14,20 @@ description = "Backtracking register allocator inspired from IonMonkey"
 repository = "https://github.com/bytecodealliance/regalloc2"
 
 [dependencies]
-log = { version = "0.4.8", default-features = false }
-smallvec = { version = "1.6.1", features = ["union"] }
-rustc-hash = { version = "1.1.0", default-features = false }
-slice-group-by = { version = "0.3.0", default-features = false }
+log = { version = "0.4.22", default-features = false }
+smallvec = { version = "1.13.2", features = ["union"] }
+rustc-hash = { version = "2.0.0", default-features = false }
```
Comment (bjorn3) [inline Cargo.toml:19]: This the only change that matters. Updating minor versions in Cargo.toml.is only necessary when you are depending on new features. Cargo will pick the latest version either way (in the absence of a lockfile pinning it to an earlier version)

Comment (JonasKruckenberg) [reply Cargo.toml:19]: Sorry yeah, just did a full `cargo upgrade` without giving it more thought 😅 updated the PR to only bump the `rustc-hash` crate

**Review by cfallin — APPROVED** [review]
Thanks!

**Comment by cfallin** [comment]
@JonasKruckenberg do you need a point-release with this update? We don't have a release schedule in regalloc2, rather we release as-needed -- happy to cut a new one for this if you like.

**Comment by cfallin** [comment]
Sure thing, will do that over in #189.

**Comment by JonasKruckenberg** [comment]
Thanks!

**Comment by cfallin** [comment]
@JonasKruckenberg this has now been [released](https://crates.io/crates/regalloc2/0.10.1) as 0.10.1.

**Pull request merged**